docs/nf-adapt.qmd

```diff
@@ -0,0 +1,107 @@
+# Adaptation with Normalizing Flows
+
+**Experimental and subject to change**
+
+Normalizing flow adaptation through Fisher HMC is a new sampling algorithm that
+automatically reparameterizes a model. It adds some computational cost outside
+model log-density evaluations, but allows sampling from much more difficult
+posterior distributions. For models with expensive log-density evaluations, the
+normalizing flow adaptation can also be much faster, if it can reduce the number
+of log-density evaluations needed to reach a given effective sample size.
+
+The normalizing flow adaptation works by learning a transformation of the parameter
+space that makes the posterior distribution more amenable to sampling. This is done
+by fitting a sequence of invertible transformations (the "flow") that maps the
+original parameter space to a space where the posterior is closer to a standard
+normal distribution. The flow is trained during warmup.
+
+For more information about the algorithm, see the paper todo
+
+Currently, a lot of time is spent on compiling various parts of the normalizing
+flow, and for small models this can take a large amount of the total time.
+Hopefully, we will be able to reduce this overhead in the future.
```

Comment:
Can a recommendation be made as to the size of model that warrants consideration of using NF?

Reply:
This doesn't really depend on the size of the model.
If it is easy to sample, using a NF isn't worth it, but you can run it on small or large models.